src/core/shape/2d_primitives.js

```diff
@@ -618,17 +618,16 @@ p5.prototype.rect = function() {
  */
 p5.prototype.square = function(x, y, s, tl, tr, br, bl) {
   p5._validateParameters('square', arguments);
-  return this._renderRect.apply(this, arguments);
+  // duplicate width for height as only one value given
+  let args = Array.prototype.slice.call(arguments, 0, 3);
+  args.push(arguments[2]);
+  args = args.concat(Array.prototype.slice.call(arguments, 4));
+  return this._renderRect.apply(this, args);
```

Comment:
Wouldn't it be easier to just do 
```
return this._renderRect(x, y, s, s, tl, tr, br, bl);
```
or if `apply` is preferred: 
```
return this._renderRect.apply(this, [x, y, s, s, tl, tr, br, bl]);
```

Reply:
Done this to just insert an element in the third position but that might not be necessary. Correcting in a while.